### PR TITLE
fix: allow input field to have zero value

### PR DESCRIPTION
**Pull request by flaminic**
### Description

This fixes changing number fields with value 0 to having value ""

### Checklist

-   [ ] API docs are generated
-   [x] Tests were added
-   [ ] Storybook demos were added

_All points above should be relevant for feature PRs. For bugfixes, some points might not be relevant. In that case, just check them anyway to signal the work is done._

---

### Screenshots

_supporting text_

### CODE REVIEW

File: components/input/src/input-field/input-field.js [hunk 1]
```diff
@@ -66,7 +66,13 @@ class InputField extends React.Component {
...
                         name={name}
                         type={type}
-                        value={value || ''}
+                        value={
+                            value === null ||
+                            value === undefined ||
+                            Number.isNaN(value)
+                                ? ''
+                                : value
+                        }
```
Comment (amcgee) [inline components/input/src/input-field/input-field.js:75]: Could this just use a [nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing)?  Are we also trying to handle `NaN` [falsy values](https://developer.mozilla.org/en-US/docs/Glossary/Falsy) or just trying to make sure `0` can be passed?

```suggestion
                        value={value ?? ''}
```

**Review by amcgee — CHANGES_REQUESTED** [review]

File: components/input/src/input-field/input-field.js (hunk 1 above)
Comment (flaminic) [reply components/input/src/input-field/input-field.js:75]: Yes, i was not sure here. I doubt NaN is ever going to be passed as a value, but I thought maybe safer to keep the previous behaviour for everything that is not 0, -0 or other forms of 0? But happy to change it as well. Do you think `??` would be better?

**Comment by amcgee** [comment]
I'm fine with either here @flaminic !

The one important thing I think is that if we're intending to have defined behavior for NaN values (the solution as written now) then we should make sure we have a unit test for that case. 

I think it would be fine to

**Review by KaiVandivier — APPROVED** [review]

**Review by amcgee — APPROVED** [review]
🦖 🎉

**Pull request closed without merging**